### DISCUSSION TITLE
How (1) prefetched data is stored (2) client-cache is used

### DISCUSSION THREAD

**Discussion by myc724** [juicedata/juicefs#701] [Q&A] [answered]
We are trying to understand format/mount options for tuning our JuiceFS setup for some applications. Could you let us know if any of our statement is in-correct? Thanks!

By default, …
- “juicefs format” create a volume with 4MB block-size
- “juicefs mount” configure a mount point as …
   - cache-size 1MB
   - buffer-size 300MB
   - prefetch 1 block in parallel 

Every read “prefetch 1 block”, so it brings in at least 4MB of data. Then, JuiceFS client 
- store the “in-use” key-value pairs in the 1MB cache.
- store the "rest of" prefetched data in the 300MB read/write buffer.

**Comment by davies** [comment, marked as answer]
By default, …

> “juicefs format” create a volume with 4MB block-size

Yes

> “juicefs mount” configure a mount point as …
> cache-size 1MB

The default cache-size is 1GB.

> buffer-size 300MB
> prefetch 1 block in parallel

Yes

> Every read “prefetch 1 block”, so it brings in at least 4MB of data. Then, JuiceFS client
> store the “in-use” key-value pairs in the 1MB cache.

1GB

> store the "rest of" prefetched data in the 300MB read/write buffer.

`cache-size` is the total amount of data cached in disk (under /var/jfsCache/xxx/).  
`buffer-size` is the amount of memory used for read/write.

**Reply by myc724** [reply]
My mistake. The default cache size is 1GB. Thanks!
